File: C#/TransportClientFX/transport/client/StartRpcClient.cs
using System;
using System.Configuration;
using System.Net.Mime;
using System.Windows.Forms;
using proiect.view;
using TransportServices.transport.services;
using TransportClient.transport.client;
using TransportNetworking.transport.network.rpcprotocol;


namespace TransportClient.transport.client
{
    public class StartRpcClient
    {
        private static int defaultChatPort = 55555;
        private static string defaultServer = "localhost";

        public void Start()
        {
            Console.WriteLine("In start");

            string serverIP = ConfigurationManager.AppSettings["TransportServer.Host"] ?? defaultServer;
            int serverPort = defaultChatPort;

            try
            {
                serverPort = int.Parse(ConfigurationManager.AppSettings["TransportServer.Port"]);
            }
            catch (FormatException ex)
            {
                Console.Error.WriteLine("Wrong port number " + ex.Message);
                Console.WriteLine("Using default port: " + defaultChatPort);
            }
            Console.WriteLine("Using server IP " + serverIP);
            Console.WriteLine("Using server port " + serverPort);

            ITransportServices server = new TransportServicesRpcProxy(serverIP, serverPort);
            //ITransportServices server = new TransportServicesJsonProxy(serverIP, serverPort);
            
            Login loginForm = new Login();
            loginForm.SetServer(server);

            Angajat mainForm = new Angajat(server);

            loginForm.SetMainForm(mainForm);
            Application.Run(loginForm);
        }
    }
}

File: Java/TransportClient/src/main/java/transport/client/StartRpcClient.java
package transport.client;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;


import transport.client.controllers.LoginController;
import transport.client.controllers.MainController;
import transport.network.rpcprotocol.TransportServiceRpcProxy;
import transport.services.ITransportServices;


import java.io.IOException;
import java.util.Properties;

public class StartRpcClient extends Application {
    private Stage primaryStage;

    private static int defaultChatPort = 55555;
    private static String defaultServer = "localhost";


    public void start(Stage primaryStage) throws Exception {
        System.out.println("In start");
        Properties clientProps = new Properties();
        try {
            clientProps.load(StartRpcClient.class.getResourceAsStream("/transportclient.properties"));
            System.out.println("Client properties set. ");
            clientProps.list(System.out);
        } catch (IOException e) {
            System.err.println("Cannot find transportclient.properties " + e);
            return;
        }
        String serverIP = clientProps.getProperty("transport.server.host", defaultServer);
        int serverPort = defaultChatPort;

        try {
            serverPort = Integer.parseInt(clientProps.getProperty("transport.server.port"));
        } catch (NumberFormatException ex) {
            System.err.println("Wrong port number " + ex.getMessage());
            System.out.println("Using default port: " + defaultChatPort);
        }
        System.out.println("Using server IP " + serverIP);
        System.out.println("Using server port " + serverPort);

        ITransportServices server = TransportServiceRpcProxy.getInstance(serverIP, serverPort);

        System.out.println("Server started");

        FXMLLoader loader = new FXMLLoader(
                getClass().getResource("/view/login.fxml"));
        Parent root=loader.load();


        LoginController ctrl =
                loader.<LoginController>getController();
        ctrl.setServer(server);


        FXMLLoader cloader = new FXMLLoader(
                getClass().getResource("/view/client.fxml"));
        Parent croot=cloader.load();


        MainController mainController =
                cloader.<MainController>getController();
        mainController.setServer(server);

        ctrl.setMainController(mainController);
        ctrl.setParent(croot);

        primaryStage.setTitle("MPP transport app");
        primaryStage.setScene(new Scene(root));
        ctrl.setStage(primaryStage);
        primaryStage.show();

    }


}